### PR TITLE
LPD-36248 | Playwright Test Fix

### PR DESCRIPTION
**Pull request by rafaellajordao**
:warning:   [Testray Task](https://testray.liferay.com/web/testray#/testflow/10254734/subtasks/10254744)

### Motivation:
Tests failing because there is no tearDown or setUp, leaving "garbage" in the other tests and causing the tests ahead to break.

### Proposed Solution:
Create variables that will be used in all tests and add a beforeEach and afterEach to clean up any test mess.

More details: [LPD-36248](https://liferay.atlassian.net/browse/LPD-36248)

### CODE REVIEW

**Comment by rafaellajordao** [comment]
ci:test:sf

**Comment by rafaellajordao** [comment]
ci:test:analytics-cloud

**Comment by rafaellajordao** [comment]
ci:test:sf

**Comment by rafaellajordao** [comment]
ci:test:analytics-cloud

File: modules/test/playwright/tests/osb-faro-web/page.spec.ts [hunk 1]
```diff
@@ -81,12 +81,15 @@ const goToWithReferrer = async function ({
...
-		tag: '@LRAC-8112 Legacy',
-	},
-
+const randomString = getRandomString();
+const siteName = 'My Site ' + randomString;
+const channelName = 'My Property ' + randomString;
+const pageName = 'My Page';
+let site;
+let project;
+let channel;
```
Comment (marcosapmf) [inline modules/test/playwright/tests/osb-faro-web/page.spec.ts:90]: @rafaellajordao can you sort the variables?

File: modules/test/playwright/tests/osb-faro-web/page.spec.ts [hunk 2]
```diff
@@ -81,40 +81,60 @@ const goToWithReferrer = async function ({
 	}, url);
 };
 
+const randomString = getRandomString();
+const siteName = 'My Site ' + randomString;
+const channelName = 'My Property ' + randomString;
+const pageName = 'My Page';
```
Comment (marcosapmf) [inline modules/test/playwright/tests/osb-faro-web/page.spec.ts:87]: @rafaellajordao can you rename to `pageTitle`?

File: modules/test/playwright/tests/osb-faro-web/page.spec.ts [hunk 3]
```diff
@@ -1024,22 +931,17 @@ test(
...
-		});
-
-		await test.step('Delete pages created in DXP during automation execution', async () => {
-			await page.goto(liferayConfig.environment.baseUrl);
 
-			await apiHelpers.jsonWebServicesLayout.deleteLayout(
-				String(sitePage.id)
+			await expect(page.getByText('1', {exact: true}).nth(2)).toBeVisible(
+				{
+					timeout: 100 * 1000,
```
Comment (marcosapmf) [inline modules/test/playwright/tests/osb-faro-web/page.spec.ts:943]: @rafaellajordao do we really need a large timeout here? If not can you remove this? The default timeout is already 15s :smile:

File: modules/test/playwright/tests/osb-faro-web/page.spec.ts [hunk 4]
```diff
@@ -1052,42 +954,32 @@ test(
...
-		const sitePage1 = await createSitePage({
-			apiHelpers,
-			pageTitle: pageTitle1,
-		});
-		const pageTitle2 = 'My Page 2';
-		const sitePage2 = await createSitePage({
-			apiHelpers,
-			pageTitle: pageTitle2,
-		});
+		const pageName2 = 'My Page 2';
```
Comment (marcosapmf) [inline modules/test/playwright/tests/osb-faro-web/page.spec.ts:957]: @rafaellajordao please rename. Bare in mind that only makes sense to have `pageName2`  if you have a  `pageName1` :wink:

**Review by marcosapmf — CHANGES_REQUESTED** [review]
Nice work @rafaellajordao :tada: 
There are just minor SF changes I suggested. Please see my comments below

**Comment by rafaellajordao** [comment]
ci:test:sf

**Comment by rafaellajordao** [comment]
ci:test:analytics-cloud

**Comment by rafaellajordao** [comment]
ci:test:sf

File: modules/test/playwright/tests/osb-faro-web/page.spec.ts [hunk 5]
```diff
@@ -246,26 +239,8 @@ test(
...
-		await test.step('Connect the DXP to AC', async () => {
-			await syncAnalyticsCloud({
-				apiHelpers,
-				channelName,
-				page,
-				siteName,
-			});
-		});
+		const blogName = 'My Blog ' + randomString;
+		const blogPage = 'My Blog Page ' + randomString;
```
Comment (marcosapmf) [inline modules/test/playwright/tests/osb-faro-web/page.spec.ts:243]: @rafaellajordao, does that make sense we rename it as `blogPageTitle`?

File: modules/test/playwright/tests/osb-faro-web/page.spec.ts [hunk 6]
```diff
@@ -1068,42 +943,32 @@ test(
...
-		const sitePage1 = await createSitePage({
-			apiHelpers,
-			pageTitle: pageTitle1,
-		});
-		const pageTitle2 = 'My Page 2';
-		const sitePage2 = await createSitePage({
-			apiHelpers,
-			pageTitle: pageTitle2,
-		});
+		const pageName = 'My Page 2';
```
Comment (marcosapmf) [inline modules/test/playwright/tests/osb-faro-web/page.spec.ts:946]: @rafaellajordao, `pageName` can we rename this to match to `pageTitleXXXX` pattern?

File: modules/test/playwright/tests/osb-faro-web/page.spec.ts [hunk 7]
```diff
@@ -1068,42 +943,32 @@ test(
...
+			apiHelpers,
+			pageTitle: pageName,
+			siteName,
 		});
 
 		await test.step('Go to My Page 1', async () => {
 			await navigateToSitePage({
 				page,
-				pageName: pageTitle1,
+				pageName: pageTitle,
```
Comment (marcosapmf) [inline modules/test/playwright/tests/osb-faro-web/page.spec.ts:957]: @rafaellajordao, it would be much effort to change this `pageName` attribute to be `pageTitle`, if it would, we can do that in another pull request :smile:

**Review by marcosapmf — CHANGES_REQUESTED** [review]
@rafaellajordao almost there :sweat_smile:

File: modules/test/playwright/tests/osb-faro-web/page.spec.ts (hunk 5 above)
Comment (rafaellajordao) [reply modules/test/playwright/tests/osb-faro-web/page.spec.ts:243]: yeahh I'll fix it!

File: modules/test/playwright/tests/osb-faro-web/page.spec.ts (hunk 7 above)
Comment (rafaellajordao) [reply modules/test/playwright/tests/osb-faro-web/page.spec.ts:957]: i'll open a ticket to work on it! [LPD-36814](https://liferay.atlassian.net/browse/LPD-36814)

**Comment by rafaellajordao** [comment]
ci:test:sf

**Comment by rafaellajordao** [comment]
ci:test:analytics-cloud

**Comment by rafaellajordao** [comment]
> **Jenkins Build:**[test-portal-acceptance-pullrequest(master)#17944](https://test-1-37.liferay.com/job/test-portal-acceptance-pullrequest(master)/17944/)
> **Jenkins Report:**[jenkins-report.html](https://test-1-37.liferay.com/userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/17944/jenkins-report.html)
> **Jenkins Suite:**analytics-cloud
> **Pull Request:**[liferay-ac#1847](https://github.com/liferay-ac/liferay-portal/pull/1847)
> **Testray Routine:**[EE Pull Request](https://testray.liferay.com/#/project/630146/routines/630264)
> **Testray Build:**[[master] ci:test:analytics-cloud - rafaellajordao > liferay-ac - PR#1847 - 2024-09-18[05:30:02]](https://testray.liferay.com/#/project/630146/routines/630264/build/14604705)
> **Testray Build ID:**14604705
> **Testray Importer:**[publish-testray-report#38847](https://test-1-37.liferay.com/job/publish-testray-report/38847/)
all modified tests are passing correctly! :smile:

**Review by marcosapmf — APPROVED** [review]
Great work @rafaellajordao :tada:

**Comment by rafaellajordao** [comment]
ci:forward
